tests/mean_04.cc

```diff
@@ -0,0 +1,125 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The SampleFlow library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of SampleFlow.
+//
+// ---------------------------------------------------------------------
+
+
+// Check the MeanValue consumer
+
+
+#include <iostream>
+#include <valarray>
+#include <random>
+
+#include <sampleflow/producers/metropolis_hastings.h>
+#include <sampleflow/consumers/mean_value.h>
+
+
+class MyTriangle
+
+{
+  public:
+
+    std::valarray<double> side_lengths;
+
+    MyTriangle()
+    {
+      side_lengths = {std::numeric_limits<double>::signaling_NaN(),
+                      std::numeric_limits<double>::signaling_NaN(),
+                      std::numeric_limits<double>::signaling_NaN()
+                     };
+    }
+
+    MyTriangle(const std::valarray<double> lengths)
```

Comment:
You could do the same here, with both the data type and the initialization. In any case, take the argument by-reference to avoid a copy.